Mathlib/NumberTheory/NumberField/CMField.lean

```diff
@@ -130,11 +148,79 @@ variable (K) in
 The complex conjugation generates the Galois group of `K/F`.
 -/
 theorem zpowers_complexConj_eq_top :
-    Subgroup.zpowers (complexConj F K) = ⊤ := by
+    Subgroup.zpowers (complexConj F K : K ≃ₐ[F] K) = ⊤ := by
   refine Subgroup.eq_top_of_card_eq _ ?_
   rw [Nat.card_zpowers, orderOf_complexConj, IsGalois.card_aut_eq_finrank,
     IsQuadraticExtension.finrank_eq_two]
 
+/--
+An element of `K` is fixed by the complex conjugation iff it comes from `F`.
+-/
+theorem complexConj_eq_self_iff (x : K) :
+    complexConj F K x = x ↔ x ∈ (algebraMap F K).range := by
+  convert (IntermediateField.mem_fixedField_iff (⊤ : Subgroup (K ≃ₐ[F] K)) x).symm using 1
+  · rw [← zpowers_complexConj_eq_top, Subgroup.forall_mem_zpowers]
+    exact (MulAction.mem_fixedBy_zpowers_iff_mem_fixedBy
+      (g := (complexConj F K : K ≃ₐ[F] K)) (a := x)).symm
+  · rw [IsGalois.fixedField_top, IntermediateField.mem_bot, RingHom.mem_range, Set.mem_range]
+
+/--
+An element of `𝓞 K` is fixed by the complex conjugation iff it comes from `𝓞 F`.
+-/
+theorem ringOfIntegersComplexConj_eq_self_iff (x : 𝓞 K) :
+    ringOfIntegersComplexConj F K x = x ↔ x ∈ (algebraMap (𝓞 F) (𝓞 K)).range := by
+  rw [← RingOfIntegers.eq_iff, coe_ringOfIntegersComplexConj, complexConj_eq_self_iff,
+    RingOfIntegers.coe_eq_algebraMap, RingHom.mem_range, RingHom.mem_range]
+  refine ⟨fun ⟨a, ha⟩ ↦ ⟨⟨a, ?_⟩, RingOfIntegers.eq_iff.mp ha⟩, ?_⟩
+  · exact (isIntegral_algebraMap_iff
+      (FaithfulSMul.algebraMap_injective F K)).mp (ha ▸ RingOfIntegers.isIntegral_coe x)
+  · rintro ⟨a, rfl⟩
+    exact ⟨a, rfl⟩
+noncomputable section units
+
+open NumberField.Units
+
+/--
+The complex conjugation as a `RingEquiv` on the group of units.
+-/
+def unitsComplexConj : (𝓞 K)ˣ ≃* (𝓞 K)ˣ :=
+  Units.mapEquiv (ringOfIntegersComplexConj F K).toMulEquiv
+
+@[simp]
+theorem coe_unitsComplexConj (x : (𝓞 K)ˣ) :
+    (unitsComplexConj F x : 𝓞 K) = ringOfIntegersComplexConj F K (x : 𝓞 K) := rfl
+
+theorem unitsComplexEmbedding_complexConj (φ : K →+* ℂ) (u : (𝓞 K)ˣ) :
+    Units.complexEmbedding φ (unitsComplexConj F u) = (Units.map (starRingEnd ℂ))
+      (Units.complexEmbedding φ u) := by
+  simp [Units.ext_iff]
+
+@[simp]
+theorem unitsComplexConj_torsion (ζ : torsion K) :
```

Comment:
Is this theorem the best way of stating in mathlib "the conjugate of a root of unit is its inverse"? If it is it deserves a docstring, if there is another way we should mention the other declaration.

Reply:
I can add another spelling that may be more familiar: 
```lean
/--
The image of a root of unity by the complex conjugation is its inverse.
-/
theorem complexConj_of_mem_torsion {ζ : (𝓞 K)ˣ} (hζ : ζ ∈ torsion K) :
    complexConj F K ζ = ζ⁻¹ := by
  simpa using congr_arg ((↑) : (𝓞 K)ˣ → K) <| unitsComplexConj_torsion F ⟨ζ, hζ⟩
```

Reply:
I think mentionning [Complex.conj_rootsOfUnity](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/RootsOfUnity/Complex.html#Complex.conj_rootsOfUnity) is enough.